llvm/lib/Target/AMDGPU/SIRegisterInfo.cpp

```diff
@@ -4182,6 +4182,21 @@ const TargetRegisterClass *SIRegisterInfo::getVGPR64Class() const {
                                 : &AMDGPU::VReg_64RegClass;
 }
 
+MachineBasicBlock::iterator
+SIRegisterInfo::getDomVRegDefInBasicBlock(Register Reg, MachineBasicBlock &MBB,
+                                          MachineBasicBlock::iterator I,
+                                          const TargetRegisterInfo *TRI) {
+  if (I == MBB.begin())
+    return MBB.end();
+  // Iterate backwards from I (exclusive) to the beginning of the basic block
+  do {
+    --I;
+    if (I->definesRegister(Reg, TRI))
```

Comment:
Remove the additional argument TRI passed to the function. TRI can be null here. Like we talked about earlier, we see the full definition of the register in the instructions we are interested in. Or, in the worst case, you can pass this pointer, which is itself a TargetRegisterInfo*.